File: karrab309-maker/clinova/flutter_patient_app/lib/models/user_model.dart
import '../utils/media_url.dart';

class UserModel {
  final int id;
  final String name;
  final String username;
  final String email;
  final String role;
  final String? locale;
  final String? profilePhotoUrl;

  UserModel({
    required this.id,
    required this.name,
    required this.username,
    required this.email,
    required this.role,
    this.locale,
    this.profilePhotoUrl,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      id: json['id'] as int,
      name: json['name'] as String? ?? '',
      username: json['username'] as String? ?? '',
      email: json['email'] as String? ?? '',
      role: json['role'] as String? ?? 'Patient',
      locale: json['locale'] as String?,
      profilePhotoUrl: resolveApiPublicUrl(json['profile_photo_url'] as String?),
    );
  }
}

File: karrab309-maker/clinova/angular-app/src/app/core/models/user.model.ts
export interface User {
  id: number;
  name: string;
  username: string;
  email: string;
  role: 'Admin' | 'Doctor' | 'Nurse' | 'Secretary' | 'Patient' | 'Laboratory' | 'Accountant';
  /** Spécialité (comptes médecin), renseignée à la création par l’admin. */
  specialty?: string | null;
  locale?: 'en' | 'fr' | 'ar' | null;
  profile_photo_path?: string | null;
  profile_photo_url?: string | null;

  /** Statut temps réel (API /doctors) */
  availability_status?: 'available' | 'busy' | 'offline' | 'on_call' | string;
  availability_last_seen_at?: string | null;
  active_patients_count?: number;
}
